### PR TITLE
Add share button modal

### PR DESCRIPTION
**Pull request by sbtorres**
This PR adds a modal that pops up when the user clicks on the "Share" button. The modal contains the following: 
- Header
- Short Description of Listing (hardcoded fake data for now)
- Links to share on:
  - Facebook (Non-functional)
  - Twitter (Non-functional)
  - Email (Non-functional)
  - Facebook Messenger (Non-functional)
- Link to copy URL (Non-functional)
- Link to embed listing in a website. (Non-functional)

Note that the icons are not perfectly formatted at this time. I will go back in the future and fix these up. I also plan to add fake listing descriptions to my database in the future. I will also move the styled components into the style.js file once I merge the branch that created that file with the master branch.

@michelledhuang23 please review! Thanks!

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -16,6 +16,21 @@ <h1>Carebnb</h1>
       from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a> 
       is licensed by <a href="http://creativecommons.org/licenses/by/3.0/" title="Creative Commons BY 3.0" target="_blank">CC 3.0 BY</a>
       <a href="http://chittagongit.com/icon/share-icon-svg-5.html">Share Icon Svg #31948</a>
```
Comment (michelledhuang23) [inline public/index.html]: http://chittagongit.com/icon/share-icon-svg-5.html seems to be a jpg not a svg. But either works!

File: client/src/ShareModal.jsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        onClick={this.handleClickOutsideModal}
+        shareModalIsVisible={shareModalIsVisible}
+      >
+        <StyledDiv ref={this.ref}>
+          <StyledCloseButton onClick={this.onCloseButtonClick}>X</StyledCloseButton>
+          <StyledHeader>
+            <StyledTitle>Share</StyledTitle>
+            <StyledListingDescription>
+              Check out this awesome listing on Carebnb:
+               Veniam voluptatem sit rerum aut sed est in suscipit.
```
Comment (michelledhuang23) [inline client/src/ShareModal.jsx]: for a second i thought i was too tired to read english lol XD

File: client/src/ShareModal.jsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  constructor(props) {
+    super(props);
+    this.ref = React.createRef();
+
+    this.onCloseButtonClick = this.onCloseButtonClick.bind(this);
+    this.handleClickOutsideModal = this.handleClickOutsideModal.bind(this);
+  }
+
+  onCloseButtonClick(event) {
+    event.preventDefault();
```
Comment (michelledhuang23) [inline client/src/ShareModal.jsx]: if event is never used is event.preventDefault() necessary?

File: public/index.html (hunk 1 above)
Comment (sbtorres) [reply public/index.html]: Yeah, luckily this one was pretty much spot on for what the icon looks like on Airbnb. I will consider going after fixing this when I cut a new branch to fix all of my svg's that are currently img tags.

File: client/src/ShareModal.jsx (hunk 3 above)
Comment (sbtorres) [reply client/src/ShareModal.jsx]: Good catch! I removed it. Thanks!

**Pull request merged**